component/common/appendable/appendable.go

```diff
@@ -37,6 +37,13 @@ func (app FlowAppendable) Appender(_ context.Context) storage.Appender {
 	}
 }
 
+// SetReceivers sets the list of receivers the appendable should forward to.
+func (app FlowAppendable) SetReceivers(receivers ...*metrics.Receiver) {
+	if receivers != nil {
+		app = FlowAppendable(receivers)
+	}
+}
```

Comment:
I don't think this will work, since it's a non-pointer receiver. Can you write a test? 

It might be better to switch back to the struct to make it easier to reason about ensuring the correct slice of receivers is being used for appending.

Reply:
The non-pointer receiver is a type alias for a pointer-type though, so that's why I thought it'd work 🤷 

In any case, I'd also be in favor of returning to a the struct, I've got a PR ready.

Reply:
> The non-pointer receiver is a type alias for a pointer-type though,

It still makes a copy of the slice header. So you have an extra value pointing to a location in memory containing a slice, and by doing `app = FlowAppendable(receivers)`, you are just changing where that new copy is pointing, not changing the slice.

Reply:
I see, both this and the next comment make sense now, sorry for not catching this in the first place.

Reply:
no worries :)